### PR TITLE
Task: add NumFOCUS logo and donation button

### LINKED ISSUE #9: add donation link to website
I'm pasting below the (somewhat lengthy) instructions for adding a donations link to your website: https://www.flipcause.com/secure/cause_pdetails/MzUwMQ==

**How-To Instructions:**
Copy This Code Block on Your Website

Note: You only need to do this step ONCE per website. Install once in either Header (below the <head> tag) or Footer (below the <footer> tag). This code will be used for ALL buttons and floating tabs that you create on your site.

```
<!--Flipcause Integration v3.0// Flipcause Integration Instructions: 
Install the following code block once in the website Header (after <head> tag) -->
<style>
.fc-black_overlay{ 
display:none; position: fixed; z-index:1000001; top: 0%;left: 0%;width: 100%;height: 100%;
background-color: black; filter: alpha(opacity=50); cursor:pointer; opacity:0.5; 
}
.fc-white_content {
opacity:1; display:none; margin-top: -320px; margin-left: -485px; width:970px; height:640px; 
position:fixed; top:50%; left:50%; border: none;z-index:1000002;overflow: auto;
}
.fc-main-box{
opacity:1; display:none; margin:15px auto 0 auto; width:930px; position:relative; z-index:1000003;
}
.fc-widget_close{
opacity:1; background: url(https://www.flipcause.com/assets/close-icon.png);
position:absolute; z-index:1000004; right:-16px; top:-16px; display:block; cursor:pointer; width:49px; height:49px;
}
.floating_button{
display: block; margin-top: 0px; margin-left: 0px; width:auto ; height: auto; 
position:fixed; z-index:999999; overflow: auto;
}
@keyframes backfadesin {
   from { opacity:0; }
   to {opacity:.5;}
}
@-moz-keyframes backfadesin { 
    from { opacity:0; }
    to {opacity:.5;}
}
@-webkit-keyframes backfadesin { 
    from { opacity:0; }
    to {opacity:.5;}
}
@-o-keyframes backfadesin {
    from { opacity:0; }
    to {opacity:.5;}
}
@-ms-keyframes backfadesin {
    from { opacity:0; }
    to {opacity:.5;}
}
@keyframes fadesin {
   0%{ opacity:0; }
   50%{ opacity:0; }
   75% {opacity: 0; transform: translateY(20px);}
   100% {opacity: 1; transform: translateY(0);}
}
@-moz-keyframes fadesin {
   0%{ opacity:0; }
   50%{ opacity:0; }
   75% {opacity: 0; -moz-transform: translateY(20px);}
   100% {opacity: 1; -moz-transform: translateY(0);}
}
@-webkit-keyframes fadesin {
   0%{ opacity:0; }
   50%{ opacity:0; }
   75% {opacity: 0; -webkit-transform: translateY(20px);}
   100% {opacity: 1; -webkit-transform: translateY(0);}
@-o-keyframes fadesin {
   0%{ opacity:0; }
   50%{ opacity:0; }
   75% {opacity: 0; -o-transform: translateY(20px);}
   100% {opacity: 1; -o-transform: translateY(0);}
}
@-ms-keyframes fadesin {
   0%{ opacity:0; }
   50%{ opacity:0; }
   75% {opacity: 0; -ms-transform: translateY(20px);}
   100% {opacity: 1; -ms-transform: translateY(0);}
}
</style>
<script>
function open_window(cause_id) {
var  protocol=String(document.location.protocol);
var new_url;
if( /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)){
new_url="https://www.flipcause.com/widget/"+cause_id
window.open(new_url);
}
else {
document.getElementById("fc-fade").style.display="block";
document.getElementById("fc-fade").style.webkitAnimation="backfadesin 1s";
document.getElementById("fc-fade").style.animation="backfadesin 1s";
document.getElementById("fc-fade").style.mozAnimation="backfadesin 1s";
document.getElementById("fc-light").style.display="block";
document.getElementById("fc-light").style.webkitAnimation="fadesin 1.5s";
document.getElementById("fc-light").style.animation="fadesin 1.5s";
document.getElementById("fc-light").style.mozAnimation="fadesin 1.5s";
document.getElementById("fc-main").style.display="block";
document.getElementById("fc-main").style.webkitAnimation="fadesin 1.5s";
document.getElementById("fc-main").style.animation="fadesin 1.5s";
document.getElementById("fc-main").style.mozAnimation="fadesin 1.5s";
document.getElementById("fc-close").style.display="block";
document.getElementById("fc-close").style.webkitAnimation="fadesin 1.5s";
document.getElementById("fc-close").style.animation="fadesin 1.5s";
document.getElementById("fc-close").style.mozAnimation="fadesin 1.5s";
document.getElementById("fc-myFrame").style.display="block";
document.getElementById("fc-myFrame").style.webkitAnimation="fadesin 1.5s";
document.getElementById("fc-myFrame").style.animation="fadesin 1.5s";
document.getElementById("fc-myFrame").style.mozAnimation="fadesin 1.5s";
document.getElementById("fc-myFrame").src="https://www.flipcause.com/widget/"+cause_id;
}
}
function close_window() {
document.getElementById("fc-fade").style.display="none";
document.getElementById("fc-light").style.display="none";
document.getElementById("fc-main").style.display="none";
document.getElementById("fc-close").style.display="none";
document.getElementById("fc-myFrame").style.display="none";
}</script>

<div id="fc-fade" class="fc-black_overlay" onclick="close_window()"></div>
<div id="fc-light" class="fc-white_content">
<div id="fc-main" class="fc-main-box">
<div id="fc-close" class="fc-widget_close" onclick="close_window()"> 
</div><iframe id="fc-myFrame" iframe height="580" width="925" style="border: 0; 
border-radius:5px 5px 5px 5px; box-shadow:0 0 8px rgba(0, 0, 0, 0.5);" scrolling="no" src=""></iframe></div>
</div>

<!--END Flipcause Main Integration Code-->

```

THEN

Copy This Button Code to Your Desired Webpage(s)

`<a href ="https://www.flipcause.com/secure/cause_pdetails/MzUwMQ==" target="_blank" style="background:#f5cb33; border-radius:0px 0px 0px 0px;font-weight:normal; font-family:Arial, Helvetica, sans-serif;border:none;box-shadow:none;left: 50%; margin-left:-75px;clear: both;display: block; width:150px;height:45px; line-height:2.8; position:relative; font-size:16px;text-align:center; cursor:pointer;color:#000000;text-decoration: none; z-index:1">Donate to Numpy</a>`

### PR DESCRIPTION
**Pull request by ZZhaoTireless**
Hi, I just add the NumFOCUS logo and donation button to numpy.org.

I notice that previous contributors may feel confused on this task, so I provide my solution here. If someone still would like to do this task, I am glad to give the way :)   

Here is the screenshot:

screen shot 2017-03-22 at 11 57 23 pm

Closes #9 #8

### CODE REVIEW

**Comment by pv** [comment]
Thanks, looks good to me based on the screenshot.

File: _templates/layout.html [hunk 1]
```diff
@@ -8,4 +8,142 @@
...
+
+{% block footer %}
+<div class="container">
+   <a href="http://www.numfocus.org/" style="float:left; margin-top:10px;">
+      <img src="_static/numfocus-logo-small.png">
+   </a>
+   <div class="row-fluid" style="margin-top:25px;">
+      <ul class="inline pull-left">
+         <li> © Copyright 2017 NumPy developers. </li>
+         <li> Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.5.3. </li>
```
Comment (eric-wieser) [inline _templates/layout.html]: Does this mean we have to edit this file every time we update sphinx? Can't this be inserted programaticaly?

Comment (ZZhaoTireless) [inline _templates/layout.html]: If we extend footer in the `layout.html`, the copyright information in `conf.py` will be overridden, or I could add the logo on somewhere else.

Comment (pv) [inline _templates/layout.html]: You can call `{{ super() }}` (or something like that) to get the automatic footer html.

Comment (ZZhaoTireless) [inline _templates/layout.html]: Thanks, I will take a look of that.

Comment (pv) [inline _templates/layout.html]: See http://www.sphinx-doc.org/en/stable/templating.html for ways to insert js files etc.

**Comment by ZZhaoTireless** [comment]
Hi, I have updated the implementation, and the screenshot is following:

screen shot 2017-03-23 at 4 07 46 pm

File: _static/donation.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+50%{ opacity:0; }
+75% {opacity: 0; -o-transform: translateY(20px);}
+100% {opacity: 1; -o-transform: translateY(0);}
+}
+@-ms-keyframes fadesin {
+0%{ opacity:0; }
+50%{ opacity:0; }
+75% {opacity: 0; -ms-transform: translateY(20px);}
+100% {opacity: 1; -ms-transform: translateY(0);}
+}
```
Comment (eric-wieser) [inline _static/donation.css]: Nit-pick: Trailing newline would be nice

File: index.rst [hunk 3]
```diff
@@ -40,6 +40,14 @@ fundamental array data structure), see `scipy.org <http://www.scipy.org/>`__.
 
 .. raw:: html
 
-   <div style="padding-top: 40px;text-align:center;">
-     <script type="text/javascript" src="http://www.ohloh.net/p/4894/widgets/project_partner_badge.js"></script>
+	<div style="padding-top:40px; text-align:center; margin:0px auto;">
+
+	 <div style="display: inline-block; float:left; margin-left:1em;">
+	  <a href ="https://www.flipcause.com/secure/cause_pdetails/MzUwMQ==" target="_blank" style="background:#f5cb33; border-radius:0px 0px 0px 0px;font-weight:normal; font-family:Arial, Helvetica, sans-serif;border:none;box-shadow:none; clear: both;display: block; width:150px;height:34px; line-height:2.8; font-size:13px;text-align:center; cursor:pointer;color:#000000;text-decoration: none; z-index:1">Donate to Numpy</a>
+	 </div>
```
Comment (eric-wieser) [inline index.rst]: Weird mix of tabs and spaces here

File: _static/donation.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+}
+
+function close_window() {
+    document.getElementById("fc-fade").style.display = "none";
+    document.getElementById("fc-light").style.display = "none";
+    document.getElementById("fc-main").style.display = "none";
+    document.getElementById("fc-close").style.display = "none";
+    document.getElementById("fc-myFrame").style.display = "none";
+}
```
Comment (eric-wieser) [inline _static/donation.js]: Trailing newline again

**Comment by ZZhaoTireless** [comment]
Hi, I fixed the nitpicks and updated.

File: index.rst [hunk 5]
```diff
@@ -40,6 +40,11 @@ fundamental array data structure), see `scipy.org <http://www.scipy.org/>`__.
 
 .. raw:: html
 
-   <div style="padding-top: 40px;text-align:center;">
-     <script type="text/javascript" src="http://www.ohloh.net/p/4894/widgets/project_partner_badge.js"></script>
-   </div>
+<div style="padding-top:40px; text-align:center; margin:0px auto;">
```
Comment (eric-wieser) [inline index.rst]: I think this might need to be indented to actually work, but I'm no expert at RST.

**Comment by ZZhaoTireless** [comment]
Fixed the indent issue of `index.rst`.

**Comment by eric-wieser** [comment]
Screenshot?

File: index.rst [hunk 6]
```diff
@@ -40,6 +40,11 @@ fundamental array data structure), see `scipy.org <http://www.scipy.org/>`__.
...
 .. raw:: html
 
-   <div style="padding-top: 40px;text-align:center;">
-     <script type="text/javascript" src="http://www.ohloh.net/p/4894/widgets/project_partner_badge.js"></script>
+   <div style="padding-top:40px; text-align:center; margin:0px auto;">
+     <div style="display: inline-block; float:left; margin-left:1em;">
+       <a href ="https://www.flipcause.com/secure/cause_pdetails/MzUwMQ==" target="_blank" style="background:#f5cb33; border-radius:0px 0px 0px 0px;font-weight:normal; font-family:Arial, Helvetica, sans-serif;border:none;box-shadow:none; clear: both;display: block; width:150px;height:34px; line-height:2.8; font-size:13px;text-align:center; cursor:pointer;color:#000000;text-decoration: none; z-index:1">Donate to Numpy</a>
+     </div>
+     <div style="display: inline-block;">
+       <script type="text/javascript" src="http://www.ohloh.net/p/4894/widgets/project_partner_badge.js"></script>
```
Comment (eric-wieser) [inline index.rst]: `https`, in case `numpy.org` gets a certificate in future?

**Comment by ZZhaoTireless** [comment]
Here is the screenshot:

screen shot 2017-03-23 at 4 39 49 pm

File: index.rst (hunk 6 above)
Comment (ZZhaoTireless) [inline index.rst]: Do you mean `https` in `https://www.flipcause.com/secure/cause_pdetails/MzUwMQ==`? I copy it from #9 under instruction. I have no idea if I could change it.

Comment (eric-wieser) [inline index.rst]: No, I mean change the `http` in `http://www.ohloh.net/p/4894/widgets/project_partner_badge.js` to `https`

Comment (ZZhaoTireless) [inline index.rst]: Sure. I see.

**Comment by ZZhaoTireless** [comment]
Updated `http` to `https`.

**Comment by ZZhaoTireless** [comment]
Updated the commit message. Thanks for reviewing my code!

**Comment by rgommers** [comment]
Looks good, thanks a lot @ZZhaoTireless!

And thanks Eric and Pauli for reviewing.

**Pull request merged**